sdks/java/core/src/main/java/org/apache/beam/sdk/util/BigQueryTableInserter.java

```diff
@@ -82,8 +80,8 @@
   private final TableReference defaultRef;
   private final long maxRowsPerBatch;
 
-  private static final ExecutorService executor = MoreExecutors.getExitingExecutorService(
-      (ThreadPoolExecutor) Executors.newFixedThreadPool(100), 10, TimeUnit.SECONDS);
+  private static final ExecutorService executor =
```

Comment:
If BigQueryTableInserter took in a PipelineOptions object then you would be able to get the Executor directly from GcsOptions instead of using the factory to create another instance. Alternatively, BigQueryTableInserter could take an Executor as a an argument to the constructor.

It seems as though the BigTableInserter is only constructed inside BigQueryIO and tests where PipelineOptions/GcsOptions is accessible.